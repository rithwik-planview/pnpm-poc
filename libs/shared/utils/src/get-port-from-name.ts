export const getPortFromName = (name: string, base = 3000) => {
    let hash = 0;
    for (let i = 0; i < name.length; i++) {
        hash = (hash << 5) - hash + name.charCodeAt(i);
        hash |= 0;
    }
    return base + (Math.abs(hash) % 1000);
};
