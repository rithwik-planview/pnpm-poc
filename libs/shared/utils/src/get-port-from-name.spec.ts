import { getPortFromName } from './get-port-from-name';

describe('getPortFromName', () => {
    it('should return the correct port for each remote name', () => {
        expect(getPortFromName('@unity/tools')).toMatchSnapshot();
        expect(getPortFromName('@unity/shared.ui')).toMatchSnapshot();
        expect(getPortFromName('@unity/shared.state')).toMatchSnapshot();
        expect(getPortFromName('@unity/features.feature')).toMatchSnapshot();
        expect(getPortFromName('@unity/apps.app')).toMatchSnapshot();
    });
});
