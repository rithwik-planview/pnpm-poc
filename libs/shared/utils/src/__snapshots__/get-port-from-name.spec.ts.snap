// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPortFromName should return the correct port for each remote name 1`] = `3681`;

exports[`getPortFromName should return the correct port for each remote name 2`] = `3189`;

exports[`getPortFromName should return the correct port for each remote name 3`] = `3378`;

exports[`getPortFromName should return the correct port for each remote name 4`] = `3081`;

exports[`getPortFromName should return the correct port for each remote name 5`] = `3577`;
